pkg/volume/git_repo/git_repo.go

```diff
@@ -126,10 +126,8 @@ func (b *gitRepoVolumeBuilder) SetUp() error {
 	return b.SetUpAt(b.GetPath())
 }
 
-// This is the spec for the volume that this plugin wraps.
-var wrappedVolumeSpec = &volume.Spec{
-	Volume: &api.Volume{VolumeSource: api.VolumeSource{EmptyDir: &api.EmptyDirVolumeSource{}}},
-}
+// This is the name for the volume that this plugin wraps, will be generated with random value in SetUpAt(..).
+var wrapperVolumeName string
```

Comment:
This can't be a global.  Every call to SetUpAt() will overwrite it, which means TearDown() will fail.  It has to be attached to each volume.  But then random doesn't work because Builder/Cleaner have different amounts of information.

I think it has to be determinable from the volName.  I left us an affordance here - the API enforces that volume names are DNS_LABEL formatted.  That gives us (I hope) freedome to use things like _ in the implementation (which we do in Docker container names, for example).

What if we simply named the wrapee "wrapped_" + volName ?